docs/html/forms/building-forms.md

```diff
@@ -5,4 +5,61 @@ sidebar_label: Building Forms
 sidebar_position: 1
 tags: [html, web-development, forms, building-forms]
 description: In this tutorial, you will learn how to build forms in HTML. Forms are used to collect user input on web pages, such as login forms, contact forms, and search forms.
----
\ No newline at end of file
+---
+
+HTML provides a set of elements and attributes to create interactive forms that allow users to input data. Forms are essential for collecting information from users, such as contact details, feedback, or preferences. In this guide, we'll cover the fundamental elements and attributes required to build forms in HTML.
+
+## The `<form>` Element
+The `<form>` element serves as a container for form elements such as input fields, buttons, checkboxes, etc. It defines the boundary of the form and specifies where the form data should be sent upon submission.
+
+        <form>
+            -.
+            -form elements
+            -.
+        </form>
+
+## The `<input>` element
+Input elements allow users to enter data into the form. There are various types of input elements, each catering to different data types and input methods
+
+Common input types include:
+
+        -Text: <input type="text">
+        -Password: <input type="password">
+        -Email: <input type="email">
+        -Checkbox: <input type="checkbox">
+        -Radio Buttons: <input type="radio">
+        -File: <input type="file">
+        -Date: <input type="date">
+        -Number: <input type="number">
+        -Submit Button: <input type="submit">
+
+## The `<label>` element
+The `<label>` element associates a text label with an input element, improving accessibility and user experience.
+
+Example:
+
```

Comment:
**bad practice** 

```md
<label for="username">Username:</label>
<input type="text" id="username" name="username">
```

**Good Practice**

````md
```html
<label for="username">Username:</label>
<input type="text" id="username" name="username">
```
````